lib/MusicBrainz/Server/Form/Annotation.pm

```diff
@@ -7,6 +7,17 @@ has '+name' => (default => 'edit-annotation');
 
 has_field 'text' => (
     type     => 'Text',
+    trim => { transform => sub {
+        my $string = shift;
+        # We don't want to break list formatting at the start of the annotation
+        if ($string =~ m/^\s*(?:(?: {4})|\t)[^\s]/) {
+            $string =~ s/^\s*((?:(?: {4})|\t)[^\s])/$1/;
+        } else {
+            $string =~ s/^\s+//;
+        }
```

Comment:
TBH this makes me uneasy looking at it, especially since the list formatting syntax depends on the wikitext parser being used.

Maybe we should just trim whitespace at the end, and leave the beginning of the string untouched.